File: doc/codes/python/examples/api/command/silencer.py
from autd3_core import Duration

from autd3.commands import FixedCompletionTime, FixedUpdateRate, SetSilencer

intensity = Duration.from_micros(250)
phase = Duration.from_micros(1000)
strict_mode = True
# ANCHOR: api
SetSilencer()

SetSilencer.disable()

SetSilencer(
    FixedCompletionTime(
        intensity,
        phase,
        strict_mode,
    )
)
# ANCHOR_END: api

intensity = 256
phase = 256

# ANCHOR: api
SetSilencer(FixedUpdateRate(intensity, phase))
# ANCHOR_END: api

File: doc/codes/rust/examples/api/command/silencer.rs
use core::num::NonZeroU16;
use std::time::Duration;

use autd3_rs::commands::{FixedCompletionTime, FixedUpdateRate, SetSilencer};

fn main() {
    let intensity = Duration::from_micros(250);
    let phase = Duration::from_micros(1000);
    let strict_mode = true;
    // ANCHOR: api
    SetSilencer::default();

    SetSilencer::disable();

    SetSilencer::new(FixedCompletionTime {
        intensity,
        phase,
        strict_mode,
    });
    // ANCHOR_END: api

    let intensity = NonZeroU16::new(256).unwrap();
    let phase = NonZeroU16::new(256).unwrap();

    // ANCHOR: api
    SetSilencer::new(FixedUpdateRate { intensity, phase });
    // ANCHOR_END: api
}
